diff --git rules/rbe_repo.bzl rules/rbe_repo.bzl
index de8501f..b78c241 100644
--- rules/rbe_repo.bzl
+++ rules/rbe_repo.bzl
@@ -811,6 +811,9 @@ _rbe_autoconfig = repository_rule(
                    "JAVA_HOME env var from the container. If that is not set, the rule " +
                    "will fail."),
         ),
+        "os_family": attr.string(
+            doc = ("Optional. The OS for which generate configuration."),
+        ),
         "registry": attr.string(
             doc = ("Optional. The registry to pull the container from. For example, " +
                    "marketplace.gcr.io. The default is the value for the selected " +
@@ -901,6 +904,7 @@ def rbe_autoconfig(
         java_home = None,
         tag = None,
         toolchain_config_suite_spec = default_toolchain_config_suite_spec(),
+        os_family = None,
         registry = None,
         repository = None,
         target_compatible_with = None,
@@ -1198,6 +1202,7 @@ def rbe_autoconfig(
         toolchain_config_suite_spec = toolchain_config_suite_spec_stripped,
         registry = registry,
         repository = repository,
+        os_family = os_family,
         tag = tag,
         target_compatible_with = target_compatible_with,
         use_checked_in_confs = use_checked_in_confs,
diff --git rules/rbe_repo/util.bzl rules/rbe_repo/util.bzl
index 71a0e9e..42a23ea 100644
--- rules/rbe_repo/util.bzl
+++ rules/rbe_repo/util.bzl
@@ -144,6 +144,8 @@ def os_family(ctx):
     Returns:
       Returns the name of the OS Family
     """
+    if ctx.attr.os_family:
+        return ctx.attr.os_family
     os_name = ctx.os.name.lower()
     if os_name.find("windows") != -1:
         return "Windows"
